specs/signature-specification.md

```diff
@@ -179,6 +152,85 @@ These attributes are considered unsigned with respect to the signing key that ge
+  - **`mediaType`** (*string*): This REQUIRED property contains media type of signature envelope blob. Following values are supported
+    - `application/jose+json`
+    - `application/cose`
+- **`subject`** (*descriptor*): A REQUIRED artifact descriptor referencing the signed manifest.
+- **`annotations`** (*string-string map*): This REQUIRED property contains metadata for the image manifest.
+  It is being used to store information about the signature.
+  Keys using the `io.cncf.notary` namespace are reserved for use in the Notary Project signature specification and MUST NOT be used by other specifications.
+  - **`io.cncf.notary.x509chain.thumbprint#S256`**: A REQUIRED annotation whose value contains the list of SHA-256 fingerprints of signing certificate and certificate chain (including root) used for signature generation. The list of fingerprints is present as a JSON array string, corresponding to ordered certificates in [*Certificate Chain* unsigned attribute](#unsigned-attributes) in the signature envelope. The annotation name contains the hash algorithm as a suffix (`#S256`) and can be extended to support other hashing algorithms in future.
+
+### OCI Signature Discovery
+
+The client should be able to discover all the signatures belonging to an artifact (such as image manifest) by using [OCI Distribution Referrers API][oci-distribution-referrers].
+OCI Distribution Referrers API returns a paginated list of all artifacts belonging to a target artifact (such as container images, SBoMs).
+The implementation can filter Notary Project signatures by either using OCI Distribution Referrers API or using custom logic on the client.
+Each Notary Project signature refers to a signature envelope blob.
+
+### OCI Signature Filtering
+
+An OCI artifact can have multiple signatures, implementations of the Notary Project signature specification uses annotations of the signature manifest to filter relevant signatures based on the applicable trust policy.
+The Notary Project signature manifest's `io.cncf.notary.x509chain.thumbprint#S256` annotation key MUST contain the list of SHA-256 fingerprints of certificate and certificate chain used for signing.
+
+## Blob Signatures
+
+The Notary Project facilitates signing arbitrary blobs using detached signatures. These detached signatures can be transported via any preferred medium and verified on the receiving end. A detached signature refers to a signature that is not embedded within (and therefore does not modify) the original blob, instead it consists of a detached signature envelope with a signature over the unique representation of the blob (i.e. blob's digest). The file extension can either be 'jws' or 'cose', indicating the signature envelope format.
+
+![Signature storage inside file system](../media/detached-signature-specification.svg)
+
+### Differences between OCI and Detached signatures
```

Comment:
We need to define "detached signatures" as detaching signatures means only containing the signature without the payload.

Reply:
> Detached signatures do contain the payload

My point is that this concept of "detached signatures" confuses people, especially those GPG users.

Without a formal definition, OCI signatures are also detached, aren't they?

Reply:
Renaming "detached signature" to "blob signature" to avoid confusion.